src/main/java/com/github/timm/cucumber/generate/CucumberITGeneratorByFeature.java

```diff
@@ -104,30 +111,32 @@ public void generateCucumberITFiles(final File outputDirectory,
 
 
             outputFileName = classNamingScheme.generate(file.getName());
-
             setFeatureFileLocation(file);
+            writeFile(outputDirectory);
 
-            final File outputFile = new File(outputDirectory, outputFileName + ".java");
-            FileWriter w = null;
-            try {
-                w = new FileWriter(outputFile);
-                writeContentFromTemplate(w);
-            } catch (final IOException e) {
-                throw new MojoExecutionException("Error creating file " + outputFile, e);
-            } finally {
-                if (w != null) {
-                    try {
-                        w.close();
-                    } catch (final IOException e) {
-                        // ignore
-                        System.out.println("Failed to close file: " + outputFile);
-                    }
+        }
+    }
+
+    private void writeFile(final File outputDirectory) throws MojoExecutionException {
+        final File outputFile = new File(outputDirectory, outputFileName + ".java");
+        FileWriter w = null;
+        try {
+            w = new FileWriter(outputFile);
+            writeContentFromTemplate(w);
+        } catch (final IOException e) {
+            throw new MojoExecutionException("Error creating file " + outputFile, e);
+        } finally {
+            if (w != null) {
+                try {
+                    w.close();
+                } catch (final IOException e) {
+                    // ignore
+                    System.out.println("Failed to close file: " + outputFile);
                 }
             }
-
-            fileCounter++;
-
         }
+
+        fileCounter++;
```

Comment:
Bit pointless, but made it easier to keep the CucumberITGeneratorByScenario.java in sync